ColorPicker/IkColorPicker.swift

```diff
@@ -0,0 +1,243 @@
+        {
+            for(var x: CGFloat = 0; x < imageWidth; x += imageWidth/256)
+            {
+                let rec = CGRect(x: x, y: y, width: recSize.width , height: recSize.height);
+                let color = self.colorFor(x, Y: y);
+                
+                CGContextSetFillColorWithColor(context, color.CGColor);
+                CGContextFillRect(context,rec);
+            }
+        }
+        
+        let hueImage = UIGraphicsGetImageFromCurrentImageContext();
+        UIGraphicsEndImageContext();
+
+        return hueImage;
+    }
+
+    func createBrightnessImage(baseColor:UIColor) -> UIImage
+    {
+        let imageHeight = CGFloat(brightnessPickerHeight);
+        let screenSize: CGRect = UIScreen.mainScreen().bounds
+
+        let imageWidth: CGFloat  = screenSize.width;
+        let size: CGSize = CGSize(width: imageWidth, height: CGFloat(imageHeight));
+        
+        UIGraphicsBeginImageContextWithOptions(size, false, 0);
+        let context: CGContextRef = UIGraphicsGetCurrentContext();
+        let recSize = CGSize(width:CGFloat(ceilf(Float(imageWidth/256))),
+            height:imageHeight);
+        
+        var hue:CGFloat = 0.0;
+        var saturation:CGFloat = 0.0;
+        baseColor.getHue(&hue, saturation:&saturation, brightness:nil, alpha:nil);
+        
+        for(var x: CGFloat = 0; x < imageWidth; x += imageWidth/256)
+        {
+            let rec = CGRect(x: x, y: 0, width: recSize.width , height: recSize.height);
+            let color = UIColor(hue: hue, saturation: saturation, brightness: (256-x)/256, alpha: 1);
+            
+            CGContextSetFillColorWithColor(context, color.CGColor);
+            CGContextFillRect(context,rec);
+        }
+        
+        let hueImage = UIGraphicsGetImageFromCurrentImageContext();
+        UIGraphicsEndImageContext();
+        
+        return hueImage;
+    }
+
+    func createFullColorImage(color :UIColor) -> UIImage
+    {
+        let imageHeight: CGFloat = 40;
+        let imageWidth: CGFloat  = 40;
+        let size: CGSize = CGSize(width: imageWidth, height: CGFloat(imageHeight));
+
+        UIGraphicsBeginImageContextWithOptions(size, false, 0);
+        let context: CGContextRef = UIGraphicsGetCurrentContext();
+        let rec = CGRect(x: 0, y: 0, width: size.width , height: size.height);
+
+        CGContextSetFillColorWithColor(context, color.CGColor);
+        let roundedRect = UIBezierPath(roundedRect: rec, cornerRadius: 6);
+        roundedRect.fillWithBlendMode(kCGBlendModeNormal, alpha: 1);
+        
+        let fullColorImage =  UIGraphicsGetImageFromCurrentImageContext();
+    
+        UIGraphicsEndImageContext();
+    
+        return fullColorImage;
+    }
+    
+    func loadView()
+    {
+        self.frame = CGRect(x:0,y:0,width:320,height:viewPickerHeight);
+        
+        let hueColorImage = self.createHueColorImage();
+        self.hueColorsImageView = UIImageView(image: hueColorImage);
```

Comment:
might as well add a '!' at the end of this so you don't have to force-unwrap it later on